packages/camera/camera/lib/src/camera_controller.dart

```diff
@@ -402,6 +403,12 @@ class CameraController extends ValueNotifier<CameraValue> {
       await CameraPlatform.instance.setDescriptionWhileRecording(description);
       value = value.copyWith(description: description);
     } else {
+      // dispose resources from previous camera description
+      if (_initCalled != null) {
+        await _initCalled;
+        _unawaited(CameraPlatform.instance.dispose(_cameraId));
```

Comment:
I think this should be awaited. This looks like it could cause race conditions with `_initializeWithDescription` if unawaited.

Reply:
I'm not sure of the implications of this either.  I need to manually test it out to see.  We may have made a mistake with adding this feature with the current version of the plugin. Since we have to call dispose on the camera anyway now, there is no benefit in using this method over creating a new `CameraController` besides adding more complexity to the plugin  (except when recording). Calling `setDescription` might also just require a following `setState` call so that `buildPreview` can be called with the new `cameraId`.